java-squid/src/main/java/org/sonar/java/se/SymbolicValueRelation.java

```diff
@@ -34,73 +38,154 @@ protected SymbolicValueRelation(SymbolicValue v1, SymbolicValue v2) {
     this.v2 = v2;
   }
 
-  public Boolean impliedBy(List<SymbolicValueRelation> knownRelations) {
+  @Override
+  public int compareTo(SymbolicValueRelation o) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1210)